### PR TITLE
Even clearer and more economical comparisons

### PR DESCRIPTION
**Pull request by pepa65**
Using native bash pattern matching instead of grep, using read with IFS instead of cut.

### CODE REVIEW

File: index.txt [hunk 1]
```diff
@@ -82,13 +82,11 @@ install_caddy()
 	fi
 
 	uname="$(uname)"
-	if grep -qi 'Darwin' <<<"$uname"; then
+	if [[ ${uname^^} == *DARWIN* ]]; then
```
Comment (mholt) [inline index.txt:85]: Explain the `^^` to me? Bash is so weird.

**Comment by pepa65** [comment]
As to the "Backup existing caddy" there are 2 considerations:
1. You don't want to overwrite an existing file.
2. You don't want an existing file to be used anymore.
3. You want to new binary to be used.

You're covering 2 in "Backup existing caddy".

You could be checking for 1 before you're moving the new binary to $install_path/$caddy_bin but also when you move a found existing caddy, it could be that $caddy_backup already exisits, so perhaps a `mv --backup=t` would be better.

As for 3, this is not strictly guaranteed from the attempts to set $install_path because you should be checking $PATH (I'm assuming that works the same in Android and other systems I'm not familiar with). Basically you need to be sure that the $install_path you come up with is in $PATH, so a check could be: `grep ':$install_path:' <<<":$PATH:"`
But if it's not in there, you'll want to start with what you find in $PATH...

File: index.txt (hunk 1 above)
Comment (pepa65) [reply index.txt:85]: Because you used grep -i (case insensitive), ${X^^} converts all lowercase in $X to uppercase. :-)

**Comment by mholt** [comment]
I think the main idea with backup existing caddy is to just make sure the old version (presumably working) is still available in case the user needs to do a quick rollback.

**Pull request merged**